### PR TITLE
Move traceAsyncErrors to Chronicles

### PR DESCRIPTION
**Pull request by zah**
This would make it easier to use the construct in more projects

### CODE REVIEW

File: chronicles/chronos_tools.nim [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import
+  chronos,
+  ../chronicles
+
+proc catchOrQuit*(error: Exception) =
+  if error of CatchableError:
```
Comment (arnetheduck) [inline chronicles/chronos_tools.nim:6]: this is not quite right though, chronos only puts catchables in futures, so this function is kind of pointless

Comment (zah) [reply chronicles/chronos_tools.nim:6]: I would argue that this function is right on a fundamental level. It's an accident that Chronos is putting only Defects in futures right now.

Comment (zah) [reply chronicles/chronos_tools.nim:6]: https://github.com/status-im/nim-chronos/issues/81#issuecomment-595288809

Comment (stefantalpalaru) [reply chronicles/chronos_tools.nim:6]: > chronos only puts catchables in futures

And `CancelledError`.

**Pull request merged**